sky/data/sky_csync.py

```diff
@@ -0,0 +1,337 @@
+    """Builds sync command for aws s3"""
+    config_cmd = ('aws configure set default.s3.max_concurrent_requests '
+                  f'{num_threads}')
+    subprocess.check_output(config_cmd, shell=True)
+    sync_cmd = f'aws s3 sync {src_path} s3://{dst}'
+    if delete:
+        sync_cmd += ' --delete'
+    if no_follow_symlinks:
+        sync_cmd += ' --no-follow-symlinks'
+    return sync_cmd
+
+
+def get_gcs_upload_cmd(src_path: str, dst: str, num_threads: int, delete: bool,
+                       no_follow_symlinks: bool):
+    """Builds sync command for gcp gcs"""
+    sync_cmd = (f'gsutil -m -o \'GSUtil:parallel_thread_count={num_threads}\' '
+                'rsync -r')
+    if delete:
+        sync_cmd += ' -d'
+    if no_follow_symlinks:
+        sync_cmd += ' -e'
+    sync_cmd += f' {src_path} gs://{dst}'
+    return sync_cmd
+
+
+def run_sync(src: str, storetype: str, dst: str, num_threads: int,
+             interval_seconds: int, delete: bool, no_follow_symlinks: bool,
+             csync_pid: int):
+    """Runs the sync command to from src to storetype bucket"""
+    #TODO(Doyoung): add enum type class to handle storetypes
+    storetype = storetype.lower()
+    if storetype == 's3':
+        sync_cmd = get_s3_upload_cmd(src, dst, num_threads, delete,
+                                     no_follow_symlinks)
+    elif storetype == 'gcs':
+        sync_cmd = get_gcs_upload_cmd(src, dst, num_threads, delete,
+                                      no_follow_symlinks)
+    else:
+        raise ValueError(f'Unsupported store type: {storetype}')
+
+    max_retries = 10
+    # interval_seconds/2 is heuristically determined
+    # as initial backoff
+    initial_backoff = int(interval_seconds / 2)
+    backoff = common_utils.Backoff(initial_backoff)
+    for _ in range(max_retries):
+        try:
+            with subprocess.Popen(sync_cmd, start_new_session=True,
+                                  shell=True) as proc:
+                _set_running_csync_sync_pid(csync_pid, proc.pid)
+                if storetype == 's3':
+                    # set number of threads back to its default value
+                    config_cmd = \
+                        ('aws configure '
+                         'set default.s3.max_concurrent_requests 10')
+                    subprocess.run(config_cmd, shell=True, check=True)
+                proc.wait()
+                _set_running_csync_sync_pid(csync_pid, -1)
+        except subprocess.CalledProcessError:
+            # reset sync pid as the sync process is terminated
+            _set_running_csync_sync_pid(csync_pid, -1)
+            src_to_bucket = (f'\'{src}\' to \'{dst}\' '
+                             f'at \'{storetype}\'')
+            wait_time = backoff.current_backoff()
+            logger.warning('Encountered an error while syncing '
+                           f'{src_to_bucket}. Retrying sync '
+                           f'in {wait_time}s. {max_retries} more reattempts'
```

Comment:
```suggestion
                           f'in {wait_time}s. {max_retries-i-1} more reattempts'
```
nit, where `i` is the loop variable